skbio/io/fasta.py

```diff
@@ -557,6 +563,10 @@
 
 @register_sniffer('fasta')
 def _fasta_sniffer(fh):
+
+    if _count_blank_lines(fh) > 5:
```

Comment:
That makes sense, thanks for looking into this more. I think the way to go is have a simple utility function used by the FASTA and FASTQ sniffers that handles the blank line logic at the beginning of the file, similar to what you have now. Then `_line_generator` can be updated to have a `skip_blanks` parameter that the readers can use accordingly, depending on whether they are at the beginning of the file or in the middle of parsing records. Does that sound good to you?